core/src/main/java/com/sequenceiq/cloudbreak/service/multiaz/MultiAzCalculatorService.java

```diff
@@ -248,4 +255,118 @@ private boolean isPreferredSubnetsSpecifiedForScaling(NetworkScaleDetails networ
         return networkScaleDetails != null
                 && CollectionUtils.isNotEmpty(networkScaleDetails.getPreferredSubnetIds());
     }
+
+    public Set<GroupPlacement> prepareGroupAzMap(Map<String, String> subnetAzPairs, Set<InstanceGroup> instanceGroups) {
+        Set<String> groups = getGroups(instanceGroups);
+        Set<String> availabilityZones = getAvailabilityZones(subnetAzPairs);
+
+        if (availabilityZones.size() < NUMBER_OF_GROUPS || groups.size() != NUMBER_OF_GROUPS) {
+            return Collections.emptySet();
+        }
```

Comment:
Why is it an issue if there are more than 3 groups provided?

Reply:
As discussed, the number of availability zones can be more than 3 but groups not.